benchmark/utils/cuda_linops.hpp

```diff
@@ -682,6 +684,190 @@ class CuspGenericCoo
         // !(defined(_WIN32) || defined(__CYGWIN__))
 
 
+namespace cuda {
+namespace ilu {
+
+
+class NamedLambdaOperation : public gko::Operation {
+public:
+    NamedLambdaOperation(
+        std::string name,
+        std::function<void(std::shared_ptr<const gko::CudaExecutor>)> op)
+        : name_{std::move(name)}, op_{std::move(op)}
+    {}
+    void run(std::shared_ptr<const gko::CudaExecutor> exec) const override
+    {
+        op_(exec);
+    }
+    const char *get_name() const noexcept override { return name_.c_str(); }
+
+private:
+    std::string name_;
+    std::function<void(std::shared_ptr<const gko::CudaExecutor>)> op_;
+};
```

Comment:
This might be a candidate for a more general utility, but how do we implement it for a single executor only?